types/src/transaction/transaction_footprint.rs

```diff
@@ -0,0 +1,137 @@
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+#[cfg(any(feature = "std", test))]
+use serde::{Deserialize, Serialize};
+
+use crate::{DeployFootprint, Digest, Gas, Timestamp};
+#[cfg(any(feature = "std", test))]
+use crate::{TimeDiff, TransactionConfig, TransactionHash};
+
+#[cfg(any(feature = "std", test))]
+use crate::TransactionConfigFailure;
+
+use super::transaction_v1::TransactionV1Footprint;
+
+#[derive(Clone, Debug)]
+#[cfg_attr(
+    any(feature = "std", test),
+    derive(Serialize, Deserialize),
+    serde(deny_unknown_fields)
+)]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
+/// A footprint of a transaction.
+pub enum TransactionFootprint {
```

Comment:
Wouldn't it be more easy if we don't separate `Deploy` from `V1` footprints?
I'm thinking that we can have something like:

```suggestion
pub struct TransactionFootprint {
    /// The header of the `Transaction`.
    pub header: TransactionHeader,
    /// The estimated gas consumption of the `Transaction`.
    pub gas_estimate: Gas,
    /// The bytesrepr serialized length of the `Transaction`.
    pub size_estimate: usize,
    /// The `Transaction` category.
    pub category: TransactionCategory,
}
```

Reply:
Indeed. This should be very simple. We are concerned with total block size in bytes, overall max computation cost for the block, and max number per kind of transaction.

If a transaction being considered for inclusion in proposed block would cause us to exceed any one of those limits, we do not include that transaction.

Reply:
There is now a single footprint for both legacy Deploys and new Transactions. Commit: https://github.com/casper-network/casper-node/pull/4513/commits/59b956846875bb430f3c98e959a3b5ccce4fb0e8